Sprint 6

Darktel opened the pull request:
Запустить тест в командной строке:

Для Chrome:
`mvn clean test -Dbrowser=chrome -e`

Для Firefox:
`mvn clean test -Dbrowser=firefox -e`

По умолчанию тесты запускаются через Firefox. 
P.s. Параметров очень много при запуске тестов. Рекомендую уменьшить кол-во параметров. [Тут](https://github.com/Darktel/qa_java/compare/sprint_6?expand=1#diff-b66216267edbd294b5b802a36c39b764357f32790e064ccd9bb5c2ad66d0e019R81).

Commit:
Написал тесты, сделал паметризацию.
Поправил pom.xml.

Commit:
фикс параметра.

Commit:
дополнил README.md 224 теста 169 - провалены. (вырезал 60% тестовых параметров).

pomegranate282922 commented on src/main/java/ru/praktikum_services/qa_scooter/MainPage.java:
```diff
@@ -0,0 +1,41 @@
+
+public class MainPage {
+    private static WebDriver driver;
+    private final By accordion = By.xpath("//*[@id='root']/div/div[1]/div[5]/div[2]");
```
Нестабильный локатор. Используем не более 4х переходов 
Нужно исправить

Darktel replied:
Исправлено в Этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/ed081f820c0adca9134888bab847644daf8bf283).
Так же исправил ошибку определения 2-й кнопки заказа на странице.

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/DriverFactory.java:
```diff
@@ -0,0 +1,30 @@
+        }
+    }
+
+    public void SetupFirefox() {
```
Методы именуем camelCase 
Нужно исправить

Darktel replied:
Исправлено в этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/25a7bfee92ae438505da6ee035026cdd708b536e).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/DriverFactory.java:
```diff
@@ -0,0 +1,30 @@
+        driver = new org.openqa.selenium.chrome.ChromeDriver();
+        driver.manage().window().maximize();
+    }
+}
```
Отлично сделан выбор браузера

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestCheckTextInAccordionPanelPassed.java:
```diff
@@ -0,0 +1,60 @@
+package ru.praktikum_services.qa_scooter;
```
В пакетах используем только строчные буквы, без _ 
Нужно исправить

Darktel replied:
Исправлено в этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/0642f4661f05b1cd9cc95dbda2a6cd54ec16339f).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestCheckTextInAccordionPanelPassed.java:
```diff
@@ -0,0 +1,60 @@
+
+    private static java.util.stream.Stream<Arguments> textInAccordionPanel() {
+        return java.util.stream.Stream.of(
+                Arguments.of( "//*[@id='accordion__heading-0']", "//*[@id='accordion__panel-0']/p", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."),
```
Локаторов в тестовом классе быть не должно 
Нужно исправить

Darktel replied:
Исправлено в этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/b636af0d4b1bde664284691794eda58ae8820230).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestCheckTextInAccordionPanelPassed.java:
```diff
@@ -0,0 +1,60 @@
+
+        @AfterEach
+        public void tearDown() {
+            driverFactory.getDriver().quit();
```
Общие начала и концы всех тестовых классов удобно вынести в общий класс предок 
Можно улучшить

Darktel replied:
Вынес инициализации драйвера через Extension. [коммит](https://github.com/Darktel/qa_java/pull/6/commits/6b73951fa717775f8d55865b35e04ae0347550a3)

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,105 @@
+    @ParameterizedTest
+    @MethodSource("provideLocatorsAndData")
+    public void testOrderScooterTopButtonPassed(By selectorsButtonOrder,
+                                                String FirstName,
```
Параметры именуем camelCase 
Нужно исправить

Darktel replied:
Исправлено в этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/f7cc8a8fb6c60e60be542c833412795a3c6502ea).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,105 @@
+        orderPage.fillDateOrder(Date)
+                .selectPeriod(daysOrderPeriod)
+                .selectColorScooter(collorScooter)
+                .sendTextForCurrier("123")
```
Courier

Darktel replied:
Исправлено в этом [коммите](https://github.com/Darktel/qa_java/pull/6/commits/40be4a91c38600f5dd058b6533d7c703b62c0b4f).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,105 @@
+        orderPage.clickButtonYes();
+//        Thread.sleep(2000);
+        // Проверяем что заказ был успешно оформлен
+        assertTrue(true, String.valueOf(orderPage.checkOrderScooterVisible()));
```
Ассерт всегда истинен 
Нужно исправить

Darktel replied:
Убрал эту проверку. По условиям задания не требуется. [Коммит](https://github.com/Darktel/qa_java/pull/6/commits/bf6b8eac1163f15703f60a8c9e991ee5f70b60bc).

pomegranate282922 commented on src/test/java/ru/praktikum_services/qa_scooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,105 @@
+                                        .flatMap(data3 -> Stream.of(colorScooter)
+                                                .flatMap(data4 -> Stream.of(daysOrderPeriod)
+                                                        .map(data5 -> Arguments.of(loc, data1, data2, data3, data4, data5)))))));
+
```
Сложновато. Можно упростить, передавая кейсы в другом порядке 
Можно улучшить

Darktel replied:
Улучшил как было указано. Сократил кол-во тестовых данных до 6-й. [коммит](https://github.com/Darktel/qa_java/pull/6/commits/6b73951fa717775f8d55865b35e04ae0347550a3)

_p.s. В этот коммит попало исправление по выносу инициализации драйвера через Extension._

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Применена параметризация. Здорово сделан выбор браузера.
замечания:
1. Нейминг
2. Убрать локаторы из тестовых классов 
3. Сделать более стабильные локаторы 
4. Поправить ассерт

Commit:
Исправил локатор для определения элемента accordion.
Исправил ошибку с 2-м локатором кнопки заказать.

Commit:
Исправил наименование методов. привел их к camelCase.

Commit:
убрал дифисы c имен пакетов. (Если не правильно нужно прояснить как?)

Commit:
Перенес локаторы c парамтров для автотеста в MainPage.java

Commit:
Исправил наименование параметров для теста. Привел их в camelCase.

Commit:
Исправил опечатку в имени метода.

Commit:
убрал лишнюю проверку для проверки условий успешности тест-кейса она не требуется.
p.s. я ее ставил при отладке, когда не доходило до этого шага по какой-то причине.

pomegranate282922 commented on src/test/java/ru/praktikumservices/qascooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,102 @@
+                                        .flatMap(data3 -> Stream.of(colorScooter)
+                                                .flatMap(data4 -> Stream.of(daysOrderPeriod)
+                                                        .map(data5 -> Arguments.of(loc, data1, data2, data3, data4, data5)))))));
+
```
java.util.stream.Stream.of(
                Arguments.of(mainPage.getButtonOrderScooterTop(), “Александр», «Гусев» и тд), … 
То бишь аналогично тому как сделано в тестах на faq.

Commit:
Изменил параметризацию теста проверки заказа.
Изменил инициализацию драйвера. Сделал ее через Extension.
Изменил версию зависимостей в pom.xml

pomegranate282922 commented on src/main/java/ru/praktikumservices/qascooter/OrderPage.java:
```diff
@@ -0,0 +1,151 @@
+    private final By buttonNextFormOrder = By.xpath("//button[@class='Button_Button__ra12g Button_Middle__1CSJM']");
+    // поля формы 1
+    // Поле ввода "Имя"
+    private final By inputFirstName = By.xpath("//*[@id='root']/div/div[2]/div[2]/div[1]/input");
```
Не все локаторы исправлены 
Нужно исправить

Darktel replied:
исправил [Тут](https://github.com/Darktel/qa_java/pull/6/commits/95f81a79a4ac2a5a902237939bc23716e9675d6c).

pomegranate282922 commented on src/test/java/ru/praktikumservices/qascooter/TestCheckTextInAccordionPanelPassed.java:
```diff
@@ -0,0 +1,55 @@
+
+    @ParameterizedTest
+    @MethodSource("textInAccordionPanel")
+    public void testCheckTextInAccordionPanelPassed(By locator_heading, By locator_paragraph, String text)  {
```
Параметры именуем camelCase 
Нужно исправить

Darktel replied:
исправил [Тут](https://github.com/Darktel/qa_java/pull/6/commits/ccb085a359abd305b38475cae747ac39612c31cb).

pomegranate282922 commented on src/test/java/ru/praktikumservices/qascooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,99 @@
+    @ParameterizedTest
+    @MethodSource("provideLocatorsAndData")
+    public void testOrderScooterTopButtonPassed(By selectorsButtonOrder,
+                                                String FirstName,
```
camelCase

Darktel replied:
исправил [Тут](https://github.com/Darktel/qa_java/pull/6/commits/ccb085a359abd305b38475cae747ac39612c31cb).

pomegranate282922 reviewed (commented):
Многое исправил - супер! Но еще немного доделать.

Commit:
Исправил все локаторы в OrderPage

Commit:
Исправил именование переменных.

Darktel commented:
Спасибо большое, что так досконально проверяете проект. 
Очень сильно этого не хватает на данном обучении.

pomegranate282922 commented on src/test/java/ru/praktikumservices/qascooter/TestCheckTextInAccordionPanelPassed.java:
```diff
@@ -0,0 +1,55 @@
+        new WebDriverWait(driver, Duration.ofSeconds(4)).until(ExpectedConditions.visibilityOfElementLocated(locatorHeading));
+        driver.findElement(locatorHeading).click();
+        new WebDriverWait(driver, Duration.ofSeconds(4)).until(ExpectedConditions.visibilityOfElementLocated(locatorParagraph));
+        assert driver.findElement(locatorParagraph).isDisplayed();
```
Используем ассерты из junit5 
Нужно исправить

Darktel replied:
исправил в [коммите](https://github.com/Darktel/qa_java/pull/6/commits/0cd9d10e448634614db34cb962081bf4f54f3ff1).

pomegranate282922 commented on src/test/java/ru/praktikumservices/qascooter/TestOrderScooterTopButtonPassed.java:
```diff
@@ -0,0 +1,98 @@
+    public void testOrderScooterTopButtonPassed(By selectorsButtonOrder,
+                                                String firstName,
+                                                String lastName,
+                                                String Date,
```
camelCase…

Darktel replied:
Надеюсь все доправил. 
Я подключил линтер в IDE заставил его тесты провалидировать... Поэтому в коммите много мелких правок, помимо указанных.

pomegranate282922 reviewed (commented):
привет! Две помарки прошу еще доделать. 
код небольшой, поэтому вполне по силам весь его просмотреть и привести в соответствие ожиданиям.

Commit:
Исправил именование переменных.
Исправил assert - переделал junit assert.

pomegranate282922 reviewed (approved):
Отлично, зачет!
Здорово, что подключил линтер!
